lib/rules/init-declarations.js

```diff
@@ -114,6 +135,15 @@ module.exports = {
 				const kind = node.kind,
 					declarations = node.declarations;
 
+				if (mode === MODE_ALWAYS) {
+					if (node.declare) {
+						return;
+					}
+					if (isAncestorNamespaceDeclared(node)) {
```

Comment:
Can we add a `TSModuleDeclaration` visitor that sets a flag that the code is inside it to avoid searching up to the root?

Reply:
Yes, updated.